the-collab-lab/tcl-44-smart-shopping-list
# Connected App To Firebase

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

I set up a database on the Firestore and linked it to the App so that data could be received and sent to the database.

### **Display items from database**

- A new collection called `List1` was created on the firestore
- Two new documents, `List1` and` List2,` were created, each having the field `Item`
- The database is linked to the App, so when you launch it, the values of the fields from the collection's documents are displayed.

### **Write items to the database**

- The app's` Add list` button, when clicked, adds a new item to the database in the sequence "item" + id. (The id starts at 0 and increases by 1 with each click of the button.)




## Related Issue
closes #16


## Acceptance Criteria

- [ ] Making a change in the Firestore database will update the data in the app
- [ ] Making a change in the app will update the data in Firestore



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|  ✓ | :link: Update dependencies |
 



## Testing Steps / QA Criteria



- From your terminal, pull down this branch with `git pull origin dd-to-connect-to-firebase` and check that branch out with `git checkout dd-to-connect-to-firebase`
- Then `npm update` to install the newly added dependencies locally and `npm start` to launch the app.
- Test to check if a new item is added to the existing list by clicking the `Add list` button.  Also, go [**here** ](https://console.firebase.google.com/u/0/project/tcl-44-smart-shopping-list/firestore/data/~2FList1~2FList2)to see if the new item has been added to the database.
